Amplify/Categories/API/ClientBehavior/AmplifyAPICategory+GraphQLBehavior+Combine.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// No-listener versions of the public APIs, to clean call sites that use Combine
+// publishers to get results
```

Comment:
ps: one thing that just came to my mind is, JS Promises emits warnings when no listeners are attached to a promise, that could be an interesting DX optimization for `nil` listeners that weren't resolved anywhere (either by the developer, or by a Combine publisher, or by a PromiseKit Promise)... Just food for thought! :)

Reply:
> I am a little bit confused by this strategy. I understand why the listener is required, but making it optional only on iOS 13 is no guarantee that the developer will consume the result through the Combine publisher, is that fair assumption?

That is correct.

> Can we provide these extensions passing `nil` by default regardless of the iOS version? I believe this would be beneficial for other 3rd party integrations as well, such as PromiseKit. Otherwise we, or the community, will have to write the same extensions for PromiseKit support.
> 
> What are your thoughts on this?

We definitely could. It would mean that customers who don't support iOS 13 would be able to execute operations with no listener, just like Combine-flavored APIs. It would simplify the API surface and bring parity to the party, too, so that's a plus.

The main question would be whether we feel comfortable with giving customers the ability to invoke without a listener, but I'm thinking that might be OK. We've made a pretty strong commitment to the paradigm that "the work being performed by an operation" is separate from "the reporting of the results of the operation", by separating cancellation of the work from cancellation of the subscription.

I think I'm in favor of this and will update.

Reply:
Force-push includes a rebase/squash, and giving default values to nil-listener APIs for all platforms, not just iOS 13.